### PR TITLE
Refactor Expr

### PR DESCRIPTION
**Pull request by kaihsin**
1) moving Hamiltonian/StepHamiltonian and ThreadedMatrix to Operator module 
2) moving all linalg associate to 1) to Operator module.

### CODE REVIEW

File: lib/BloqadeExpr/src/Operator/Operator.jl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    export Hamiltonian, StepHamiltonian
+    export ThreadedMatrix
+    export set_backend
+    export storage_size, to_matrix
+
+    include("types.jl")
+    include("printings.jl")
+    include("linalg.jl")
+    include("preferences.jl")
```
Comment (Roger-luo) [inline lib/BloqadeExpr/src/Operator/Operator.jl:21]: no need to indent.

File: lib/BloqadeExpr/src/Operator/Operator.jl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module Operator
```
Comment (Roger-luo) [inline lib/BloqadeExpr/src/Operator/Operator.jl:1]: Operator sounds like a confusing name with YaoBlocks's staff, let's use a more restrictive word for this? maybe just `LowLevel`?

Comment (kaihsin) [reply lib/BloqadeExpr/src/Operator/Operator.jl]: This is changed in the new commit. module rename Operator -> Lowlevel

File: lib/BloqadeExpr/Project.toml [hunk 3]
```diff
@@ -5,6 +5,7 @@ version = "0.1.13"
 [deps]
 Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
 BitBasis = "50ba71b6-fa0f-514d-ae9a-0916efc90dcf"
+BloqadeKrylov = "bd27d05e-4cd1-5e79-84dd-c5d7d508ade2"
```
Comment (weinbe58) [inline lib/BloqadeExpr/Project.toml:8]: Do we need this dependency?

File: lib/BloqadeExpr/Project.toml [hunk 4]
```diff
@@ -28,9 +29,9 @@ LaTeXStrings = "1"
 LuxurySparse = "0.7"
```
Comment (weinbe58) [inline lib/BloqadeExpr/Project.toml:29]: If we need BloqadeKrylov we need to update compatability.

File: lib/BloqadeExpr/src/Lowlevel/linalg.jl [hunk 5]
```diff
@@ -0,0 +1,79 @@
+#**************************************************************
+#  Here, one can find the linalg API for  
+#  1. Hamiltonian/StepHamiltonian/ValHamiltonian 
+#  2. Binding of standard LinearAlgebra functions 
+#     to the backend of choice for ThreadedMatrix
+#**************************************************************
```
Comment (weinbe58) [inline lib/BloqadeExpr/src/Lowlevel/linalg.jl:6]:
```suggestion
#**************************************************************
#  Here, one can find the linalg API for  
#  1. Hamiltonian/SumOfLinop
#  2. Binding of standard LinearAlgebra functions 
#     to the backend of choice for ThreadedMatrix
#**************************************************************
```

File: lib/BloqadeExpr/src/Lowlevel/types.jl [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    end
+    )
+end
+
+## convert StepHamiltonian to ValHamiltonian 
+function ValH(h::StepHamiltonian)
+    # get values of fs:
+    return ValHamiltonian(get_f(h),h.h)
+end
+=#
```
Comment (weinbe58) [inline lib/BloqadeExpr/src/Lowlevel/types.jl:163]: Do we want to keep this here?

**Review by weinbe58 — APPROVED** [review]
LGTM after addressing the comments.

File: lib/BloqadeExpr/Project.toml (hunk 3 above)
Comment (kaihsin) [reply lib/BloqadeExpr/Project.toml:8]: I don't think we need that I will remove it

File: lib/BloqadeExpr/src/Lowlevel/types.jl (hunk 6 above)
Comment (kaihsin) [reply lib/BloqadeExpr/src/Lowlevel/types.jl]: I removed it

**Pull request merged**